### PR TITLE
[video] VIDEO_UTILS::CAsyncGetItemsForPlaylist::GetItemsForPlaylist: …

### PR DESCRIPTION
**Pull request by ksooo**
…If sort by date, year, episodenum enforce sort order ascending for playlist items.

Imo, it makes absolutely no sense to sort playlist items by date, descending. Happens with PVR recording folder contents which are by default sorted by date, descending. But if you want to play the folder content, you for sure want to watch from oldest to newest.

Runtime-tested on macOS and Android, latest Kodi master.

@enen92 I hope you see my point for the change in behavior and that you can approve the code change.

### CODE REVIEW

File: xbmc/video/VideoUtils.cpp [hunk 1]
```diff
@@ -182,6 +182,8 @@ void CAsyncGetItemsForPlaylist::GetItemsForPlaylist(const std::shared_ptr<CFileI
 
       if (sortDesc.sortBy == SortByLabel)
         items.ClearSortState();
+      else if (sortDesc.sortBy == SortByDate && sortDesc.sortOrder == SortOrderDescending)
```
Comment (enen92) [inline xbmc/video/VideoUtils.cpp:185]: Can't this be defined before? It looks to have it defined as descending and rewrite the value. Ideally should be set to ascending when it is born

Comment (ksooo) [reply xbmc/video/VideoUtils.cpp:185]: Good point. I will revise the change accordingly.

Comment (ksooo) [reply xbmc/video/VideoUtils.cpp]: Should be better now, @enen92 ...

**Review by garbear — APPROVED** [review]
I approve changing a "date"-related sort method to Ascending in this particular instance.

A lot of times, when browsing my media, I sort by Date or Year, and it defaults to Descending, which I find annoying. Do you know of any other places where switching from a default Descending to a default Ascending could happen?

**Comment by ksooo** [comment]
> A lot of times, when browsing my media, I sort by Date or Year, and it defaults to Descending, which I find annoying. Do you know of any other places where switching from a default Descending to a default Ascending could happen?

No, not really. Those definitions seem to be spread all over our code base.

**Comment by ksooo** [comment]
WIN-UWP-32 Jenkins error is unrelated.

**Comment by ksooo** [comment]
jenkins build this please

**Review by enen92 — APPROVED** [review]
Thanks for taking my comment into consideration

**Pull request merged**